scripts/ci-e2e.sh

```diff
@@ -22,38 +22,35 @@
 set -o nounset
 set -o pipefail
 
-# our exit handler (trap)
-cleanup() {
-  # If Boskos is being used then release the GCP project back to Boskos.
-  [ -z "${BOSKOS_HOST:-}" ] || hack/checkin_account.py
-}
-
-trap cleanup EXIT
+BOSKOS_HOST=${BOSKOS_HOST:-"boskos.test-pods.svc.cluster.local."}
 
 REPO_ROOT=$(dirname "${BASH_SOURCE[0]}")/..
 cd "${REPO_ROOT}" || exit 1
 
+echo "using boskos host to checkout project: ${BOSKOS_HOST}"
+
 # If BOSKOS_HOST is set then acquire an GCP account from Boskos.
-if [ -n "${BOSKOS_HOST:-}" ]; then
-  # Check out the account from Boskos and store the produced environment
-  # variables in a temporary file.
-  account_env_var_file="$(mktemp)"
-  python hack/checkout_account.py 1>"${account_env_var_file}"
-  checkout_account_status="${?}"
-
-  # If the checkout process was a success then load the account's
-  # environment variables into this process.
-  # shellcheck disable=SC1090
-  [ "${checkout_account_status}" = "0" ] && . "${account_env_var_file}"
-
-  # Always remove the account environment variable file. It contains
-  # sensitive information.
-  rm -f "${account_env_var_file}"
-
-  if [ ! "${checkout_account_status}" = "0" ]; then
-    echo "error getting account from boskos" 1>&2
-    exit "${checkout_account_status}"
-  fi
+# Check out the account from Boskos and store the produced environment
+# variables in a temporary file.
+account_env_var_file="$(mktemp)"
+python hack/checkout_account.py 1>"${account_env_var_file}"
+checkout_account_status="${?}"
+
+# If the checkout process was a success then load the account's
+# environment variables into this process.
+# shellcheck disable=SC1090
+[ "${checkout_account_status}" = "0" ] && . "${account_env_var_file}"
+
+# Always remove the account environment variable file. It contains
+# sensitive information.
+rm -f "${account_env_var_file}"
+
+if [ ! "${checkout_account_status}" = "0" ]; then
+  echo "error getting account from boskos" 1>&2
+  exit "${checkout_account_status}"
 fi
 
 (cd "${REPO_ROOT}" && hack/ci/e2e-conformance.sh)
+
+echo "cleaning up - checking in boskos account $BOSKOS_RESOURCE_NAME on host $BOSKOS_HOST"
+hack/checkin_account.py
```

Comment:
Putting this back into a signal handler would probably be good, otherwise this will probably not run if the conformance test fails.

Reply:
Done! thanks @detiber